### ISSUE TITLE
Risk of division by zero error in ScrollView

### ISSUE THREAD

**Issue by brentpicasso** [kivy/kivy#7006]
**Software Versions**
* Python: 3.7.7
* OS: linux
* Kivy: 1.11.1
* Kivy installation method: development

**Describe the bug**
Chance of division by zero error in scrollview when resizing scroll window
https://github.com/kivy/kivy/blob/master/kivy/uix/scrollview.py#L914

This code should guard against the chance of a divide by zero condition, under any circumstance. 

**Expected behavior**
No division by zero exception thrown during widget resizing.

**Repro**
Repro is unpredictable, but will happen. involves a scroll view that can be resized within the app. Example exception:

```[CRITICAL] [CrashHandler] Caught exception in Kivy loop: float division by zero
[CRITICAL] [Traceback (most recent call last)]
  File "/home/brent/git-projects/RC_app_private/rc-venv/lib/python3.7/site-packages/kivy/core/window/window_sdl2.py", line 747, in mainloop
    self._mainloop()
  File "/home/brent/git-projects/RC_app_private/rc-venv/lib/python3.7/site-packages/kivy/core/window/window_sdl2.py", line 479, in _mainloop
    EventLoop.idle()
  File "/home/brent/git-projects/RC_app_private/rc-venv/lib/python3.7/site-packages/kivy/base.py", line 342, in idle
    self.dispatch_input()
  File "/home/brent/git-projects/RC_app_private/rc-venv/lib/python3.7/site-packages/kivy/base.py", line 327, in dispatch_input
    post_dispatch_input(*pop(0))
  File "/home/brent/git-projects/RC_app_private/rc-venv/lib/python3.7/site-packages/kivy/base.py", line 286, in post_dispatch_input
    wid.dispatch('on_touch_move', me)
  File "kivy/_event.pyx", line 707, in kivy._event.EventDispatcher.dispatch
  File "/home/brent/git-projects/RC_app_private/rc-venv/lib/python3.7/site-packages/kivy/uix/scrollview.py", line 801, in on_touch_move
    if self.dispatch('on_scroll_move', touch):
  File "kivy/_event.pyx", line 707, in kivy._event.EventDispatcher.dispatch
  File "/home/brent/git-projects/RC_app_private/rc-venv/lib/python3.7/site-packages/kivy/uix/scrollview.py", line 864, in on_scroll_move
    dy = touch.dy / float(height - height * self.vbar[1])
ZeroDivisionError: float division by zero
```

**Comment by DanTheMan2000** [comment]
I just had the same ZeroDivisionError when scrolling through a recycled grid layout:
```
 Traceback (most recent call last):
   File "C:/Users/dseiler/Desktop/android/branches/RP3/Application/main.py", line 493, in <module>
     MainApp().run()
   File "C:\python-venv37\lib\site-packages\kivy\app.py", line 950, in run
     runTouchApp()
   File "C:\python-venv37\lib\site-packages\kivy\base.py", line 573, in runTouchApp
     EventLoop.mainloop()
   File "C:\python-venv37\lib\site-packages\kivy\base.py", line 347, in mainloop
     self.idle()
   File "C:\python-venv37\lib\site-packages\kivy\base.py", line 391, in idle
     self.dispatch_input()
   File "C:\python-venv37\lib\site-packages\kivy\base.py", line 342, in dispatch_input
     post_dispatch_input(*pop(0))
   File "C:\python-venv37\lib\site-packages\kivy\base.py", line 301, in post_dispatch_input
     wid.dispatch('on_touch_move', me)
   File "kivy\_event.pyx", line 709, in kivy._event.EventDispatcher.dispatch
   File "C:\python-venv37\lib\site-packages\kivy\uix\scrollview.py", line 843, in on_touch_move
     if self.dispatch('on_scroll_move', touch):
   File "kivy\_event.pyx", line 709, in kivy._event.EventDispatcher.dispatch
   File "C:\python-venv37\lib\site-packages\kivy\uix\scrollview.py", line 890, in on_scroll_move
     dx = touch.dx / float(width - width * self.hbar[1])
 ZeroDivisionError: float division by zero
```

**Comment by calne-ca** [comment]
I just had the same error in an empty RecycleView. The error is being raised by the ScrollView when using the "bars" scroll_type and the size of the ScrollView's content is small enough so that no scrollbar is being displayed (for example if it contains an empty layout). In this case if you drag on the place where the scroll bar would normally be it raises the ZeroDivisionError.
Code to reproduce:
```python
from kivy.app import App
from kivy.lang import Builder
from kivy.uix.scrollview import ScrollView

Builder.load_string('''
<ScrollView>:
    bar_width: 10
    scroll_type: ['bars']
    BoxLayout:
        orientation: 'vertical'
''')

class TestApp(App):
    def build(self):
        return ScrollView()

if __name__ == '__main__':
    TestApp().run()
```

**Comment by bgschaid** [comment]
I just had the same bug. The problem is in scrollview.py: in the on_scroll_move-method there is a guard against this problem for hbar
```Python
                if touch.ud.get('in_bar_x', False):
                    if self.hbar[1] != 1:
                        dx = touch.dx / float(width - width * self.hbar[1])
                        self.scroll_x = min(max(self.scroll_x + dx, 0.), 1.)
                        self._trigger_update_from_scroll()
```
while vbar there is no guarding condition
```Python
                if touch.ud.get('in_bar_y', False):
                    dy = touch.dy / float(height - height * self.vbar[1])
                    self.scroll_y = min(max(self.scroll_y + dy, 0.), 1.)
                    self._trigger_update_from_scroll()
```
Adding a condition 
`if self.vbar[1] != 1` 
there fixes the problem

**Comment by misl6** [comment]
Fixed via #7814
